hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheRecordStore.java

```diff
@@ -162,53 +164,62 @@ public AbstractCacheRecordStore(String cacheNameWithPrefix, int partitionId, Nod
+                throw new IllegalStateException("Eviction config cannot be null!");
+            }
+            this.wanReplicationEnabled = cacheService.isWanReplicationEnabled(cacheNameWithPrefix);
+            this.disablePerEntryInvalidationEvents = cacheConfig.isDisablePerEntryInvalidationEvents();
+            if (cacheConfig.isStatisticsEnabled()) {
+                statistics = cacheService.createCacheStatIfAbsent(cacheNameWithPrefix);
+            }
+            if (cacheConfig.getCacheLoaderFactory() != null) {
+                Factory<CacheLoader> cacheLoaderFactory = cacheConfig.getCacheLoaderFactory();
+                injectDependencies(cacheLoaderFactory);
+                cacheLoader = cacheLoaderFactory.create();
+                injectDependencies(cacheLoader);
+            }
+            if (cacheConfig.getCacheWriterFactory() != null) {
+                Factory<CacheWriter> cacheWriterFactory = cacheConfig.getCacheWriterFactory();
+                injectDependencies(cacheWriterFactory);
+                cacheWriter = cacheWriterFactory.create();
+                injectDependencies(cacheWriter);
+            }
+            if (cacheConfig.getExpiryPolicyFactory() != null) {
+                Factory<ExpiryPolicy> expiryPolicyFactory = cacheConfig.getExpiryPolicyFactory();
+                injectDependencies(expiryPolicyFactory);
+                defaultExpiryPolicy = expiryPolicyFactory.create();
+                injectDependencies(defaultExpiryPolicy);
+            } else {
+                throw new IllegalStateException("Expiry policy factory cannot be null!");
+            }
+
+            this.cacheContext = cacheService.getOrCreateCacheContext(cacheNameWithPrefix);
+            this.records = createRecordCacheMap();
+            this.evictionChecker = createCacheEvictionChecker(evictionConfig.getSize(), evictionConfig.getMaximumSizePolicy());
+            this.evictionPolicyEvaluator = createEvictionPolicyEvaluator(evictionConfig);
+            this.evictionStrategy = createEvictionStrategy(evictionConfig);
+            this.objectNamespace = CacheService.getObjectNamespace(cacheNameWithPrefix);
+            this.persistWanReplicatedData = canPersistWanReplicatedData(cacheConfig, nodeEngine);
+            this.cacheRecordFactory = new CacheRecordFactory(cacheConfig.getInMemoryFormat(), ss);
+            this.valueComparator = getValueComparatorOf(cacheConfig.getInMemoryFormat());
+            this.clearExpiredRecordsTask = cacheService.getExpirationManager().getTask();
+
+            injectDependencies(evictionPolicyEvaluator.getEvictionPolicyComparator());
+            registerResourceIfItIsClosable(cacheWriter);
+            registerResourceIfItIsClosable(cacheLoader);
+            registerResourceIfItIsClosable(defaultExpiryPolicy);
+            init();
+        } finally {
+            try {
+                tenantContext.close();
+            } catch (IOException ex) {
+                throw new RuntimeException(ex);
```

Comment:
use `throw ExceptionUtil.rethrow(ex);` instead here. If `ex` is already a runtime exception, it will be thrown as is, otherwise it will be wrapped in a `HazelcastException`.